### DISCUSSION TITLE
Pocketbase error: "TypeError: Failed to fetch"

### DISCUSSION THREAD

**Discussion by Nils19024** [pocketbase/pocketbase#4697] [Q&A]
I started a new Pocketbase instance on my windows11 PC and using it with NextJS 14

```
2024/04/03 14:52:46 Server started at http://127.0.0.1:8090
├─ REST API: http://127.0.0.1:8090/api/
└─ Admin UI: http://127.0.0.1:8090/_/
```

I tried to connect and get data from a client component in react:

```
"use client"

import PocketBase from 'pocketbase';

 useEffect(() => {
    async function test() {
      try {
        const POCKET_BASE_URL = "http://192.168.230.28:8090";
        //const POCKET_BASE_URL = "http://127.0.0.1:8090";
        const pb = new PocketBase(POCKET_BASE_URL)
        const cookie = Cookies.get('pb_auth');
        pb.authStore.loadFromCookie(cookie);
        const records = await pb.collection('user').getFullList({});
      } catch (error) {
        console.log(error.originalError)
      }
    }

    test()
  }, [])
```

When i use my IP adress of my PC `http://192.168.230.28:8090` i get the error:

```
TypeError: Failed to fetch
    at Client.send (pocketbase.es.mjs:20:32093)
    at RecordService.getList (pocketbase.es.mjs:20:8112)
    at RecordService.getList (pocketbase.es.mjs:20:19692)
    at request (pocketbase.es.mjs:20:9402)
    at RecordService._getFullList (pocketbase.es.mjs:20:9514)
    at RecordService.getFullList (pocketbase.es.mjs:20:7966)
    at RecordService.getFullList (pocketbase.es.mjs:20:19638)
    at test (page.jsx:38:53)
    at eval (page.jsx:45:5)
```

It only works with the localhost IP adress: `http://127.0.0.1:8090`. But when i am using the localhost adress i can only access the database with my PC. But i want to acces the database with my phone too (it is in the same network as my PC). Can anyone help me with this problem? Thanks a lot!

**Comment by ganigeorgiev** [comment]
You can try binding to all IPv4 interfaces with the `--http` flag, ex:
```sh
./pocketbase serve --http='0.0.0.0:8090'
```

If that doesn't work you can try using a tunneling service like [ngrok](https://ngrok.com/), [zrok](https://zrok.io/) or similar that will generate a public https address accessible from everywhere.

_Side-note: I noticed that you've submitted several Next.js related questions. If you are new to web development I don't recommend using PocketBase with Next.js (or any JS SSR framework) as it comes with a lot of complications. If you are using Next.js in a server-side context you already have a backend and may not need PocketBase at all (for some ramblings related to this check https://github.com/pocketbase/pocketbase/discussions/4412#discussioncomment-8575834, https://github.com/pocketbase/pocketbase/discussions/3792#discussioncomment-7620032 and probably others)._
